jaxopt/_src/cvxpy_wrapper.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""CVXPY wrappers."""
+
+from typing import Any
+from typing import Callable
+from typing import Optional
+from typing import Tuple
+
+from dataclasses import dataclass
+
+import jax
+import jax.numpy as jnp 
+from jaxopt._src import base
+from jaxopt._src import implicit_diff as idf
+from jaxopt._src import linear_solve
+from jaxopt._src import tree_util
+
+
+def _check_params(params_obj, params_eq=None, params_ineq=None):
```

Comment:
One way to factor some code would be to do:
`from jaxopt._src.eq_qp import _check_params as _check_params_eq`